pkg/validation/internal/community.go

```diff
@@ -13,30 +14,48 @@ import (
 	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
 )
 
-// IndexImagePathKey defines the key which can be used by its consumers
-// to inform where their index image path is to be checked
-const IndexImagePathKey = "index-path"
+// FilePathKey defines the key which can be used by its consumers
+// to inform where their index bundle image or annotations path to be checked
+const FilePathKey = "file"
 
-// ocpLabelindex defines the OCP label which allow configure the OCP versions
+// RangeKey defines the key which can be used by its consumers
+// to inform where their label range value only to be checked
+const RangeKey = "range"
+
+// ocpLabel defines the OCP label which allow configure the OCP versions
 // where the bundle will be distributed
-const ocpLabelindex = "com.redhat.openshift.versions"
+const ocpLabel = "com.redhat.openshift.versions"
+
+// deprecateOcpLabelMsg1_22 returns the specific ocp label message which is valid only for 1.22/OCP 4.9
+const deprecateOcpLabelMsg1_22 = "this bundle is using APIs which were deprecated and " +
+	"removed in v1.22. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22. " +
+	"Migrate the APIs " +
+	"for %s or provide compatible version(s) via the labels. (e.g. LABEL %s='4.6-4.8')"
 
-// CommunityOperatorValidator validates the bundle manifests against the required criteria to publish
-// the projects on the community operators
+// AlphaOCPCommunityOperatorValidator validates the bundle manifests against the required criteria to publish
```

Comment:
Why are we specifically referencing openshift in this repository?

Reply:
We discussed this subject in the past remember. We needed to create a validator to test the max OCP version which is only valid for OCP and it fails in the OCP community rules which seems to be a grey area.  

We have not been away so far to provide validators that could be used by any tool via a lib that address requirements more like vendor-specific. So, we agreed (https://github.com/operator-framework/api/pull/120) to move forward with. That seems a discussion where we have been trying to find a solution for like 2 yrs. 

However, one of the feedbacks were about the name and we make it clear that it is to check the criteria against  ocp community catalog. Also, it seems that we also need to make clear that it is alpha and our intention is to move it out. See that it is highlighted via this PR in the code and docs that will be generated for this validator. The other changes in this pr are based on the feedbacks as well.

a) Improve the logic to check the versions with the @joelanford suggestions
b) allow informing the annotations or the dockerfile when we want to check the OCP config label. 
c) allow to pass a range too and improve the coverage. 

Also, in the future, now that we have at least a need to address we could try to use it to solve this matter and decouple this test from here to another place. So, IHMO we have:

**Problem:** We do have not an API that can be used to address needs that are vendor-specific / OCP

**Suggested Solution:**

Decouple the operator-framework/api and creates a project that will respect its interfaces and only address checks which are vendor-specific  

Use Cases

- I am as a maintainer would like to be able to write custom checks and validators vendor-specific that respects the same interfaces used for the others in a specific project so that, we can easily ensure that operator-framework api has no vendor-specific logics as desirable
- I am as a maintainer would like to have the vendor-specific implementations also available to be used as a LIB in any project such as SDK, audit or OPM so that, I can easily keep the implementation maintained with an encapsulated logic and useful for any project.

Reply:
Hi @kevinrizza I came back for the original name and keep alpha.